### ISSUE TITLE
💅 `lint/correctness/noUnusedFunctionParameters` false positive

### ISSUE THREAD

**Issue by ev3nvy** [biomejs/biome#4002]
### Environment information

```bash
CLI:
  Version:                      1.9.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.10.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/9.9.0"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Linter:
  JavaScript enabled:           true
  JSON enabled:                 true
  CSS enabled:                  true
  GraphQL enabled:              false
  Recommended:                  false
  All:                          true
  Enabled rules:
  suspicious/noCatchAssign
  complexity/useLiteralKeys
  complexity/noMultipleSpacesInRegularExpressionLiterals
  suspicious/useNamespaceKeyword
  complexity/noUselessEmptyExport
  nursery/useConsistentMemberAccessibility
  suspicious/noAssignInExpressions
  suspicious/noDuplicateSelectorsKeyframeBlock
  suspicious/noDuplicateParameters
  correctness/noConstructorReturn
  style/useSelfClosingElements
  suspicious/noMisplacedAssertion
  style/useTemplate
  correctness/noUnusedLabels
  correctness/noUnreachableSuper
  suspicious/noCompareNegZero
  a11y/noAutofocus
  correctness/noUnsafeOptionalChaining
  correctness/noConstAssign
  suspicious/noControlCharactersInRegex
  style/noVar
  suspicious/noDoubleEquals
  suspicious/noEmptyInterface
  suspicious/noConstEnum
  nursery/noDuplicateElseIf
  correctness/noPrecisionLoss
  nursery/noRestrictedImports
  correctness/noSetterReturn
  correctness/noInvalidConstructorSuper
  suspicious/noImplicitAnyLet
  a11y/useKeyWithClickEvents
  suspicious/noDuplicateObjectKeys
  suspicious/noUnsafeDeclarationMerging
  suspicious/noReactSpecificProps
  correctness/noInnerDeclarations
  correctness/useArrayLiterals
  style/noUselessElse
  correctness/noInvalidDirectionInLinearGradient
  nursery/noValueAtRule
  nursery/noMissingVarFunction
  suspicious/noImportantInKeyframe
  complexity/noUselessLabel
  complexity/noUselessCatch
  a11y/useAriaPropsForRole
  style/useCollapsedElseIf
  correctness/noNonoctalDecimalEscape
  suspicious/noDuplicateTestHooks
  style/noDoneCallback
  complexity/noStaticOnlyClass
  complexity/noUselessUndefinedInitialization
  a11y/noInteractiveElementToNoninteractiveRole
  suspicious/noLabelVar
  correctness/noUnnecessaryContinue
  suspicious/noApproximativeNumericConstant
  nursery/noDuplicateCustomProperties
  correctness/noUnknownUnit
  correctness/noEmptyCharacterClassInRegex
  correctness/useImportExtensions
  suspicious/noSparseArray
  a11y/useIframeTitle
  nursery/noExportedImports
  a11y/noSvgWithoutTitle
  correctness/noVoidElementsWithChildren
  correctness/useJsxKeyInIterable
  style/useExportType
  complexity/noUselessLoneBlockStatements
  style/noArguments
  a11y/useValidAriaValues
  suspicious/noGlobalAssign
  suspicious/noCommentText
  suspicious/noThenProperty
  suspicious/useGetterReturn
  style/useFragmentSyntax
  a11y/noPositiveTabindex
  correctness/noRenderReturnValue
  correctness/useExhaustiveDependencies
  nursery/noSecrets
  security/noGlobalEval
  a11y/noRedundantRoles
  correctness/noUnusedVariables
  style/noYodaExpression
  suspicious/noSelfCompare
  suspicious/noAsyncPromiseExecutor
  nursery/noIrregularWhitespace
  security/noDangerouslySetInnerHtml
  style/useNodejsImportProtocol
  suspicious/noArrayIndexKey
  complexity/noWith
  suspicious/noDuplicateClassMembers
  complexity/noExtraBooleanCast
  performance/noAccumulatingSpread
  suspicious/noConfusingLabels
  correctness/noChildrenProp
  correctness/noUnknownFunction
  correctness/noInvalidPositionAtImportRule
  nursery/useDeprecatedReason
  style/noShoutyConstants
  a11y/noAriaUnsupportedElements
  nursery/noCommonJs
  correctness/noFlatMapIdentity
  a11y/noBlankTarget
  a11y/useHeadingContent
  nursery/noDynamicNamespaceImportAccess
  correctness/useValidForDirection
  correctness/noInvalidUseBeforeDeclaration
  a11y/noAriaHiddenOnFocusable
  a11y/useGenericFontNames
  correctness/noInvalidNewBuiltin
  style/noNegationElse
  complexity/useSimplifiedLogicExpression
  correctness/noUnusedFunctionParameters
  nursery/useSortedClasses
  style/noNamespaceImport
  style/noRestrictedGlobals
  correctness/noUndeclaredVariables
  nursery/noStaticElementInteractions
  nursery/useAriaPropsSupportedByRole
  nursery/useAdjacentOverloadSignatures
  style/noUnusedTemplateLiteral
  style/useExponentiationOperator
  suspicious/useAwait
  style/noNamespace
  suspicious/noDuplicateAtImportRules
  complexity/noUselessFragments
  performance/useTopLevelRegex
  complexity/noUselessStringConcat
  correctness/noUnusedImports
  suspicious/noEmptyBlock
  suspicious/noFunctionAssign
  performance/noDelete
  suspicious/noUnsafeNegation
  a11y/useValidLang
  suspicious/useValidTypeof
  a11y/useValidAriaRole
  correctness/noConstantCondition
  a11y/useAriaActivedescendantWithTabindex
  style/useExplicitLengthCheck
  style/useDefaultParameterLast
  complexity/noEmptyTypeParameters
  nursery/useComponentExportOnlyModules
  correctness/noUnknownProperty
  nursery/useTrimStartEnd
  complexity/noUselessTernary
  suspicious/noExplicitAny
  correctness/noSwitchDeclarations
  complexity/noUselessTypeConstraint
  suspicious/noRedundantUseStrict
  style/useLiteralEnumMembers
  suspicious/noGlobalIsNan
  suspicious/noSkippedTests
  suspicious/noMisleadingCharacterClass
  a11y/noLabelWithoutControl
  style/useDefaultSwitchClause
  correctness/noStringCaseMismatch
  suspicious/noRedeclare
  suspicious/noFallthroughSwitchClause
  complexity/noUselessThisAlias
  correctness/noUnreachable
  complexity/noThisInStatic
  complexity/useOptionalChain
  suspicious/noDuplicateCase
  style/noParameterAssign
  a11y/useValidAnchor
  complexity/useRegexLiterals
  correctness/noSelfAssign
  correctness/noInvalidBuiltinInstantiation
  nursery/noProcessEnv
  style/useShorthandFunctionType
  suspicious/noEmptyBlockStatements
  style/useThrowNewError
  suspicious/noShadowRestrictedNames
  a11y/useMediaCaption
  correctness/noUnsafeFinally
  style/useNodeAssertStrict
  style/useEnumInitializers
  a11y/useHtmlLang
  correctness/noUndeclaredDependencies
  nursery/noRestrictedTypes
  suspicious/noEvolvingTypes
  style/useWhile
  complexity/useArrowFunction
  style/noInferrableTypes
  a11y/noNoninteractiveTabindex
  complexity/useSimpleNumberKeys
  correctness/useYield
  style/useNumericLiterals
  suspicious/noImportAssign
  suspicious/useDefaultSwitchClauseLast
  nursery/noDuplicatedFields
  correctness/noGlobalObjectCalls
  style/noParameterProperties
  a11y/useAltText
  performance/noBarrelFile
  suspicious/noSuspiciousSemicolonInJsx
  complexity/noBannedTypes
  style/useThrowOnlyError
  suspicious/noPrototypeBuiltins
  style/useAsConstAssertion
  nursery/noEnum
  suspicious/noDebugger
  suspicious/noMisleadingInstantiator
  complexity/noVoid
  nursery/noUnknownPseudoClass
  a11y/useFocusableInteractive
  correctness/noUnmatchableAnbSelector
  suspicious/noDuplicateJsxProps
  correctness/noEmptyPattern
  complexity/noExcessiveNestedTestSuites
  performance/noReExportAll
  a11y/useKeyWithMouseEvents
  security/noDangerouslySetInnerHtmlWithChildren
  suspicious/noExtraNonNullAssertion
  nursery/noUnknownPseudoElement
  suspicious/noShorthandPropertyOverrides
  style/useConst
  style/noNonNullAssertion
  complexity/useFlatMap
  correctness/useHookAtTopLevel
  correctness/useIsNan
  suspicious/noGlobalIsFinite
  style/useConsistentBuiltinInstantiation
  suspicious/noDuplicateFontNames
  a11y/noDistractingElements
  nursery/useValidAutocomplete
  style/useConsistentArrayType
  style/useForOf
  nursery/useImportRestrictions
  nursery/useStrictMode
  a11y/useValidAriaProps
  a11y/noRedundantAlt
  complexity/useDateNow
  suspicious/noConfusingVoidType
  suspicious/noFocusedTests
  a11y/useButtonType
  a11y/useSemanticElements
  correctness/noInvalidGridAreas
  style/useFilenamingConvention
  style/useShorthandArrayType
  style/useShorthandAssign
  suspicious/useErrorMessage
  correctness/noConstantMathMinMaxClamp
  correctness/noUnusedPrivateClassMembers
  correctness/noVoidTypeReturn
  correctness/noUnknownMediaFeatureName
  nursery/noUselessEscapeInRegex
  a11y/useAnchorContent
  complexity/noUselessRename
  nursery/noSubstr
  style/useNumberNamespace
  complexity/noUselessConstructor
  a11y/noAccessKey
  complexity/noUselessSwitchCase
  style/noDefaultExport
  style/useSingleVarDeclarator
  suspicious/noExportsInTest
  suspicious/useNumberToFixedDigitsArgument
  a11y/noNoninteractiveElementToInteractiveRole
  style/noCommaOperator
  suspicious/useIsArray
  nursery/useConsistentCurlyBraces
  correctness/noNewSymbol
  a11y/noHeaderScope
  suspicious/noMisrefactoredShorthandAssign
  complexity/noForEach
  suspicious/noClassAssign
  style/noImplicitBoolean
  style/useImportType

Workspace:
  Open Documents:               0
```

### Rule name

lint/correctness/noUnusedFunctionParameters

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=YwBvAG4AcwB0ACAAdABlAHMAdAAgAD0AIAB7AAoACQBhADoAIAAiAHMAbwBtAGUAIABzAHQAcgBpAG4AZwAiACwACgAJAGIAOgAgACIAYQBuAG8AdABoAGUAcgAgAHMAdAByAGkAbgBnACIALAAKAH0AOwAKAAoAYwBvAG4AcwB0ACAAcgBlAG0AbwB2AGUAQQAgAD0AIAAoAG8AYgBqACkAIAA9AD4AIABvAGIAagAuAG0AYQBwACgAKAB7ACAAYQAsACAALgAuAC4AcgBlAHMAdAAgAH0AKQAgAD0APgAgAHIAZQBzAHQAKQA7AAoACgBjAG8AbgBzAHQAIABfAHQAZQBzAHQAMQAgAD0AIAByAGUAbQBvAHYAZQBBACgAdABlAHMAdAApADsACgA%3D

### Expected result

The lint should probably not emit a warning for this case. See [Discord discussion](https://discord.com/channels/1132231889290285117/1132231889911029825/1286364841769959485).

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
`noUnusedVariables` ignores this case. We should be able to borrow some logics from `noUnusedVariables` to fix `noUnusedFunctionParameters`.
